src/LeanCode.ContractsGenerator.Tests/ExampleBased/Properties.cs

```diff
@@ -68,4 +68,20 @@ public void Properties_with_struct_types()
             .WithProperty("A", TypeRefExtensions.Internal("InnerDTO"))
             .WithProperty("B", TypeRefExtensions.Internal("InnerDTO").Nullable());
     }
+
+    [Fact]
+    public void Properties_inside_record_types()
+    {
+        "properties/record.cs".Compiles().WithDto("DTO").WithProperty("A", Known(KnownType.Int32));
```

Comment:
I'd also check if `EqualityContract` is not propagated to results if we're to make sure it's indistinguishable from non-record classes.